src/components/Button/src/Button.vue

```diff
@@ -130,37 +144,110 @@ export default {
 			default: undefined,
 		},
 		/**
-		 * Background color of button
+		 * Main color of button
 		 */
 		color: {
 			type: String,
 			default: undefined,
 			validator: (color) => colord(color).isValid(),
 		},
 		/**
-		 * Text color of button
+		 * Main hover color of button
+		 */
+		colorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
+		/**
+		 * Text color of button (only applied on fill buttons)
 		 */
 		textColor: {
 			type: String,
 			default: undefined,
 			validator: (color) => colord(color).isValid(),
 		},
+		/**
+		 * Text hover color of button (only applied on fill buttons)
+		 */
+		textColorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
 		/**
 		 * Variant
-		 * @values primary, secondary, tertiary
+		 * @values fill, outline, ghost
 		 */
 		variant: {
 			type: String,
 			default: undefined,
-			validator: (variant) => ['primary', 'secondary', 'tertiary'].includes(variant),
+			validator: (variant) => ['fill', 'outline', 'ghost'].includes(variant),
 		},
 		/**
 		 * Shape of button
+		 * @values squared, rounded, pill, 'Npx', 'N%'
 		 */
 		shape: {
 			type: String,
 			default: undefined,
-			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape),
+			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape) || shape.match(/\d*\.?\d+(?:px|%)/i),
+		},
+		/**
+		 * Shape of button
+		 * @values 'Npx', 'N%'
+		 */
+		shapeHover: {
+			type: String,
+			default: undefined,
+			validator: (shape) => shape.match(/\d*\.?\d+(?:px|%)/i),
```

Comment:
since this regex is used in a couple places can we refactor it out and declare it as a constant above? e.g.,
```js
// matches '{number}px' & '{number}%' strings
const PX_PERCENT_REGEX = /\d*\.?\d+(?:px|%)/i;
```